### PR TITLE
Always take screenshot when share button is clicked

### PR DESCRIPTION
**Pull request by shakao**
Fixes https://github.com/microsoft/pxt-arcade/issues/1916

### CODE REVIEW

File: webapp/src/share.tsx [hunk 1]
```diff
@@ -221,15 +221,17 @@ export class ShareEditor extends data.Component<ShareEditorProps, ShareEditorSta
...
         this.setState({ recordingState: ShareRecordingState.ScreenshotSnap, recordError: undefined },
-            () => {
-                this.props.parent.requestScreenshotAsync()
-                    .then(img => {
-                        const st: ShareEditorState = { recordingState: ShareRecordingState.None, recordError: undefined };
-                        if (img) st.screenshotUri = img;
-                        else st.recordError = lf("Oops, screenshot failed. Please try again.")
-                        this.setState(st);
-                    });
+            () => this.screenshotAsync());
```
Comment (riknoll) [inline webapp/src/share.tsx:224]:
```suggestion
            this.screenshotAsync);
```

File: webapp/src/share.tsx [hunk 2]
```diff
@@ -221,15 +221,17 @@ export class ShareEditor extends data.Component<ShareEditorProps, ShareEditorSta
...
+    }
 
+    screenshotAsync = () => {
+        return this.props.parent.requestScreenshotAsync()
+            .then(img => {
+                console.log("reqd screenshot")
```
Comment (pelikhan) [inline webapp/src/share.tsx:230]: pxt.debug(...

Comment (riknoll) [inline webapp/src/share.tsx:230]: Was this log left in intentionally?

File: webapp/src/share.tsx [hunk 3]
```diff
@@ -382,7 +384,11 @@ export class ShareEditor extends data.Component<ShareEditorProps, ShareEditorSta
                 // save project name if we've made a change change
                 p = this.props.parent.updateHeaderNameAsync(newProjectName);
             }
-            p.then(() => this.props.parent.anonymousPublishAsync(screenshotUri))
+            // if screenshots are enabled, always take one
+            if (targetTheme.simScreenshot && !screenshotUri) {
+                p = p.then(() => this.screenshotAsync());
```
Comment (riknoll) [inline webapp/src/share.tsx:389]:
```suggestion
                p = p.then(this.screenshotAsync);
```

File: webapp/src/share.tsx [hunk 4]
```diff
@@ -382,7 +384,11 @@ export class ShareEditor extends data.Component<ShareEditorProps, ShareEditorSta
                 // save project name if we've made a change change
                 p = this.props.parent.updateHeaderNameAsync(newProjectName);
             }
-            p.then(() => this.props.parent.anonymousPublishAsync(screenshotUri))
+            // if screenshots are enabled, always take one
+            if (targetTheme.simScreenshot && !screenshotUri) {
```
Comment (pelikhan) [inline webapp/src/share.tsx:388]: we should not attach a screenshot "silently", the user should always be able to review it. This should be done before publishing.

File: webapp/src/share.tsx (hunk 2 above)
Comment (shakao) [reply webapp/src/share.tsx:230]: oops nope, missed it while cleaning up

**Comment by shakao** [comment]
hmm i don't want to add an extra click to the publishing flow though. this doesn't overwrite the user screenshot (or gif) if they have already taken one, this just ensures that all shared urls have SOME kind of image

**Comment by pelikhan** [comment]
Ok. Make sure things still work when sharing a project with errors (simulator not running).

**Pull request merged**